packages/tpetra/core/src/Tpetra_Details_residual.hpp

```diff
@@ -58,6 +58,79 @@
 namespace Tpetra {
   namespace Details {
 
+
+template<class ExecutionSpace>
+int64_t
+residual_launch_parameters (int64_t numRows,
```

Comment:
I'm just as guilty of copy-pasting this magic, but it would still be nice if we had it in one place.